usagov/usagov-2021
# Usagov 430 spanish directories

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-403

## Description

* Changes
To implement Spanish Directories the same view **Federal Agencies** is used but a secondary block **Indice Agencias A-Z** with *Content: Translation language set to Español* and a secondary **Alpha List ES** also with translation set to Español. Both the  **Federal Agencies A-Z** and **Indice Agencias A-Z** blocks have the link to standalone page text changed from `<a href="{{ view_node }}">More information about {{ title }}&nbsp;></a>` to only `{{ view_node }}`. This allows the rest of the link text to be set in **views-view-fields.html.twig** with the correct language following the pattern for the rest of the fields.

* The first implementation of Spanish Directories created a separate view. This methodology worked however it required  three additional Twig files (*views-view-summary-Spanish.html.twig, views-view-list-Spanish.html.twig, views-view-fields-Spanish.html.twig*) and two additional configuration files. To reduce repeat code the decision was made to utilize the same View with duplicate block. This allowed the original (*views-view-summary.html.twig, views-view-list.html.twig, views-view-fields.html.twig*) with a language check.

* Using only one block was considered for both languages however to implement this would then require a supplemental pre-process hook to only retrieve the agencies and Alpha List of the selected language. This would actually introduce more code into the repository to be maintained than the current implementation with two blocks. Though the block and alpha list or essentially duplicates of the English ones this allows them to share the same three Twig files for better maintainability.

* To Test 
    * Import config changes

    * Import federal directories following https://github.com/usagov/usagov-2021/blob/dev/web/modules/custom/usagov_directories/docs/Importing_Federal_Agency_Records.md 

    * Create a new basic page 
        * Title- Agencias federales
        * "Español" selected for language
        * URL alias- /indice-agencias
        
Spanish agencies should be displayed on the page. 
Headings in accordion should be in Spanish
Letter Indexes should take you to correct Spanish query url 
Previous and Back buttons should also  take you to correct Spanish query url 

## Checklist for the Developer



## Checklist for the Peer Reviewers
